### PR TITLE
Benchmarked model data

### PR DESCRIPTION
**Pull request by wgyn**
Allows for benchmarking models on the same set of target outcomes. The flow would be upload an "actuals" file with `id` and `actual` columns, then upload matching "scores" files with `id` and `pred_score` columns. I haven't done anything to make these display any differently since the file structure would reflect the grouping.

E.g. the titanic example would show the following models :
- data/titanic/good_model
- data/titanic/random_model

while the actuals file would live in data/titanic/actuals.tsv

Thoughts @jvns?

### CODE REVIEW

File: topmodel/plot_helpers.py [hunk 1]
```diff
@@ -39,8 +39,8 @@ def draw_labels(ax, xs, ys, thresholds, labels_left=False):
         coords = (x + 0.01, y)
         if labels_left:
             coords = (x - 0.190, y - 0.01)
-        annotation = "{threshold}: [{x},{y}] ".format(x=x_round, y=y_round,
-                                                      threshold=threshold_round)
+        annotation = "{threshold}: [{x},{y}] ".format(
+            x=x_round, y=y_round, threshold=threshold_round)
```
Comment (jvns) [inline topmodel/plot_helpers.py]: Can you separate out these style/formatting changes into a separate PR?

File: topmodel/model_data.py [hunk 2]
```diff
@@ -157,3 +156,41 @@ def get_notes(self):
...
+
+class BMModelData(ModelData):
+
+    """
+    Benchmarked model data. Store actuals in a separate 'actual.tsv' file
+    with observations identifiers in an 'id' column and actuals in an
+    'actual' column. Upload scores in a 'scores_bm.tsv' file that has
+    a matching 'id' column' and 'pred_scores' column. Throws an error
+    if the scores and actuals do not completely line up.
+    """
```
Comment (jvns) [inline topmodel/model_data.py]: :+1: thanks for the super helpful comment!

File: topmodel/model_data.py [hunk 3]
```diff
@@ -157,3 +156,41 @@ def get_notes(self):
     def set_notes(self, note):
         notes_path = os.path.join(self.model_path, NOTES_FILE)
         return self.file_system.write_file(notes_path, note)
+
+
+class BMModelData(ModelData):
```
Comment (jvns) [inline topmodel/model_data.py]: I think "BenchmarkedModelData" would be clearer here

File: topmodel/model_data.py [hunk 4]
```diff
@@ -157,3 +156,41 @@ def get_notes(self):
...
+    """
+    Benchmarked model data. Store actuals in a separate 'actual.tsv' file
+    with observations identifiers in an 'id' column and actuals in an
+    'actual' column. Upload scores in a 'scores_bm.tsv' file that has
+    a matching 'id' column' and 'pred_scores' column. Throws an error
+    if the scores and actuals do not completely line up.
+    """
+
+    def to_data_frame(self, **kwargs):
+        def indexed_data_frame(fp_or_buffer):
```
Comment (jvns) [inline topmodel/model_data.py]: can you make this function not nested?

File: topmodel/model_data.py [hunk 5]
```diff
@@ -157,3 +156,41 @@ def get_notes(self):
...
+            basedir, _ = os.path.split(self.model_path)
+            actuals_path = os.path.join(basedir, ACTUALS_FILE)
+            actuals = self.file_system.read_file(actuals_path)
+            with io.BytesIO(actuals) as f:
+                df_actuals = indexed_data_frame(f)
+
+            scores_path = os.path.join(self.model_path, SCORES_BM_FILE)
+            scores = self.file_system.read_file(scores_path)
+            with io.BytesIO(scores) as f:
+                df_scores = indexed_data_frame(f)
```
Comment (jvns) [inline topmodel/model_data.py]: There's some code duplication here that could be moved into `indexed_data_frame()`

**Comment by jvns** [comment]
This looks great overall! It's not that big of a change and I'm excited about using it. I mostly want the style/formatting changes separated out so I can understand it a little better.

**Comment by wgyn** [comment]
@jvns Made the changes you suggested and condensed to just the changes relevant to BenchmarkedModelData.

**Comment by jvns** [comment]
looks great!

**Comment by wgyn** [comment]
thanks for the review!

**Pull request merged**